.circleci/config.yml

```diff
@@ -8,23 +8,45 @@ defaults: &defaults
   docker:
     - image: circleci/node:10
 
+save_code: &save_code
+  save_cache:
+    key: code-{{ .Revision }}
+    paths:
+      - .git
+
+save_dep: &save_dep
+  save_cache:
+    key: dependency-cache-{{ checksum "yarn.lock" }}
+    paths:
+      - node_modules
+
+restore_code: &restore_code
+  restore_cache:
+    key: code-{{ .Revision }}
+
+restore_dep: &restore_dep
+  restore_cache:
+    key: dependency-cache-{{ checksum "yarn.lock" }}
+
 jobs:
   build:
-    <<: *defaults
+    working_directory: /home/circleci/project
+    # use customized docker (ubuntu, yarn 1.15, node10, solc binary)
+    docker:
+      - image: web3docker/counterfactual:0.0.1
```

Comment:
Who owns this?

Reply:
ah.. That's me... security concern?

Reply:
Could be build the `Dockerfile` in the CI process with a tag and then save the image in CircleCI cache to avoid needing to make an external image pull?